### DISCUSSION TITLE
Reproducing high visual resolution plot

### DISCUSSION THREAD

**Discussion by jddorellanao** [pyvista/pyvista#5173] [Q&A] [answered]
I was reading an [article](https://library.seg.org/share/TMSEFPGG3PVEQCB2UD6W?target=10.1190/tle42080541.1) on high visual resolution in geophysics, in which the author points that "in its simplest form, an HVR display is a three-dimensional surface with seismic amplitudes forming the terrain." 

The following images represent this HVR thing

First, we have the 2d image, which represents the amplitudes:

![Hector-3D-Scene-3-VD (Custom)](https://github.com/pyvista/pyvista/assets/86092171/6a8e8266-e4ba-437a-849f-e7afb3589fe3)

Then, we have the same data but in HVR, as you can see now the amplitudes form the terrain (Z-axis):
![Hector-3D-Scene-3-Strat (Custom)](https://github.com/pyvista/pyvista/assets/86092171/5cdce14e-67d7-4598-a1b1-bacd800c7fcd)

So I wanted to give it a try. First, the [data](https://github.com/jddorellanao/ModelosDeVelocidadesSinteticosSeismic/blob/main/49_77_PT1_PR.SGY) I used is a seismic line from Alaska.

```
import matplotlib.pyplot as plt
import numpy as np
import segyio
import pyvista as pv

with segyio.open('49_77_PT1_PR.SGY', strict=False) as s:
    ms = np.stack([t.astype(float) for t in s.trace])
s.close()

clip = abs(np.percentile(ms, 0.999))
ms_ = np.clip(ms, -clip, clip)
plt.imshow(ms_.T, cmap='seismic', aspect='auto')
plt.colorbar()
```
![image](https://github.com/pyvista/pyvista/assets/86092171/e6eda839-cf1d-42ee-b875-abe8ec566a25)

As you can see the colorbar, it does not make any sense to plot it with this range (as 3D surface). I tried to clip it from `-10` to `10`, the ImageData looks good
```
grid = pv.ImageData(dimensions=(1472, 1501, 1), spacing=(1, 1, 1), origin=(0, 0, 0))
grid.point_data["Amp"] = ms_.ravel(order="F")
``` 
![image](https://github.com/pyvista/pyvista/assets/86092171/5205f25b-b0d1-4ff6-b126-b6d546264e7c)

but the "terrain" result was not the desired one.

```
terrain = grid.warp_by_scalar()
```

![image](https://github.com/pyvista/pyvista/assets/86092171/65525cd5-c63f-4712-bd02-80f027c579c3)

Do you see it possible to reproduce what is shown in the article?

**Comment by darikg** [comment, marked as answer]
Why clip to 10? Using the original `clip = abs(np.percentile(ms, 0.999))` seems fine to me

```python
grid.warp_by_scalar(factor=20 / clip).plot(colormap='gray')
```

![image](https://github.com/pyvista/pyvista/assets/6875882/8504b335-14a5-4932-bcd8-e5cd87da197a)

You could do some smoothing first to make it look nicer:

```python
terrain = grid.gaussian_smooth(radius_factor=1.5).warp_by_scalar(factor=20 / clip)
terrain.plot(colormap='gray')
```
![image](https://github.com/pyvista/pyvista/assets/6875882/454f8e5d-5870-4b07-8669-a768f2132737)

If you convert the terrain to a PolyData you could also play around with the 3d smoothing filters

```
terrain2 = pv.PolyData(terrain.points, faces=terrain.cast_to_unstructured_grid().cells)
terrain2.point_data['Amp'] = terrain.point_data['Amp']
terrain2.smooth(n_iter=100).plot()
```

**Reply by jddorellanao** [reply]
Thank you so much! I have no idea about the `factor` parameter.
